conda_lock/src_parser/aggregation.py

```diff
@@ -41,10 +41,18 @@ def aggregate_lock_specs(
     except ValueError as e:
         raise ChannelAggregationError(*e.args)
 
+    try:
+        pip_repositories = suffix_union(
+            lock_spec.pip_repositories for lock_spec in lock_specs
+        )
```

Comment:
Yep TBH I just mimicked the logic for `channels` above. I will read the code more closely to try to understand why its needed.

Reply:
Okay so basically what this is doing is ensuring that the package sources from multiple aggregated `environment.yaml` files are compatible.

I think its basically guarding against the following situation:

- `environment-main.yaml` has `[private-package-source, other-package-source, public-package-source]`
- `environment-dev.yaml` has `[other-package-source, private-package-source, public-package-source]`

When aggregating these to dependency specification files, it is hard to say what the correct aggregated priority order for package sources should be. When resolving dependencies, should I look in `private-package-source` first, or `other-package-source` first?

Because of this, we require that the package source list in each file to be aggregated needs to be either an extension or subset of the package sources in all other source files. This means you can just use the longest package source list from all of the files, and dependency resolution will be predictable.

Reply:
This is already tested [here](https://github.com/conda/conda-lock/blob/372312eb86d1f87110ba3006566cedbcdfa239ec/tests/test_conda_lock.py#L1603-L1648).

I could extend it to check `pip-repositories` too if you like - the behaviour should be identical.

Reply:
Before adding more tests I'd like to understand https://github.com/conda/conda-lock/issues/530

Reply:
We've understood things up to possibly some edge cases in Poetry, and it seems like it mostly fits:
https://github.com/conda/conda-lock/pull/529#issuecomment-1766060611